lib/type-collection.js

```diff
@@ -63,29 +63,24 @@ module.exports = Collection.extend(lodashMixin, parentMixin, {
    * @param {Any} value - value to be added
    */
   addToType: function(value) {
-    var newTypeAdded = false;
+    var field = this.parent;
+
     var typeName = getTypeName(value);
+    var added = false;
     // get or create type
     var type = this.get(typeName);
     if (!type) {
-      type = this.add({
-        name: typeName
-      });
-      newTypeAdded = true;
+      type = this.add({name: typeName}, {sort: true});
```

Comment:
The same pattern appears in value-collection too. Wouldn't we have to make it consistent in both cases?

How is this sort on insert implemented in ampersand-collection? If it assumes a sorted array that's ok. But if it does a full sort on each insert that would be very expensive. I wanted to be on the safe side, hence the sort at serialize()